src/AppBundle/Twig/SourceCodeExtension.php

```diff
@@ -20,20 +26,47 @@
  * @author Ryan Weaver <weaverryan@gmail.com>
  * @author Javier Eguiluz <javier.eguiluz@gmail.com>
  */
-class SourceCodeExtension extends \Twig_Extension
+class SourceCodeExtension extends Twig_Extension
 {
-    protected $loader;
-    protected $controller;
-    protected $template;
-    protected $kernelRootDir;
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
 
-    public function __construct(\Twig_LoaderInterface $loader, $kernelRootDir)
-    {
+    /**
+     * @var Twig_LoaderInterface
+     */
+    private $loader;
+
+    /**
+     * @var string
+     */
+    private $kernelRootDir;
+
+    /**
+     * @var array|null
+     */
+    private $controller;
+
+    /**
+     * @param ContainerInterface $container
+     * @param Twig_LoaderInterface $loader
+     * @param $kernelRootDir
+     */
+    public function __construct(
+        ContainerInterface $container,
```

Comment:
@stof I can add a comment referring to the issue and explaining the best practice and problem.